src/firefly/js/templates/lightcurve/LcPhaseTable.js

```diff
@@ -0,0 +1,232 @@
+
+        const xyPlotParams = {
+            userSetBoundaries: {xMax: 2},
+            x: {columnOrExpr: LC.PHASE_CNAME, options: 'grid'},
+            y: {columnOrExpr: flux, options: 'grid,flip'}
+        };
+        loadXYPlot({chartId: tbl_id, tblId: tbl_id, xyPlotParams});
+    });
+
+}
+
+/**
+ * @summary adding phase column to raw table
+ * @param {string} flux flux column name
+ * @param {string} time time column name
+ * @param {string} period periodt  value
+ * @param {string} tzero zero time value
+ * @return {TableModel} phase table
+ */
+export function doPFCalculate(flux, time, period, tzero) {
+    const rawTable = getTblById(LC.RAW_TABLE);
+
+    var phaseFoldedTable = rawTable && addPhaseToTable(rawTable, time, flux, tzero, period);
+    phaseFoldedTable&&repeatDataCycle(time, parseFloat(period), phaseFoldedTable);
+
+    return phaseFoldedTable;
+}
+
+
+/**
+ * @summary calculate phase and return as text
+ * @param {number} time
+ * @param {number} timeZero
+ * @param {number} period
+ * @param {number} dec
+ * @returns {string}  folded phase (positive or negative) with 'dec'  decimal digits
+ */
+export function getPhase(time, timeZero, period,  dec=DEC_PHASE) {
+    var q = (time - timeZero)/period;
+    var p = q >= 0  ? (q - Math.floor(q)) : (q + Math.floor(-q));
+
+    return p.toFixed(dec);
+}
+
+/**
+ * @summary create a table model with phase column
+ * @param {TableModel} tbl
+ * @param {string} timeName
+ * @param {string} fluxName
+ * @param {string} tzero
+ * @param {string} period
+ * @returns {TableModel}
+ */
+function addPhaseToTable(tbl, timeName, fluxName, tzero, period) {
+    var {columns} = tbl.tableData;
+    var tIdx = timeName ? columns.findIndex((c) => (c.name === timeName)) : -1;
+
+    if (tIdx < 0) return null;
+
+    var tPF = Object.assign(cloneDeep(tbl), {tbl_id: LC.PHASE_FOLDED, title: 'Phase Folded'},
+                                            {request: getPhaseFoldingRequest(period, timeName, fluxName, tbl)},
+                                            {highlightedRow: 0});
+    tPF = omit(tPF, ['hlRowIdx', 'isFetching']);
+
+    var phaseC = {desc: 'number of period elapsed since starting time.',
+                  name: LC.PHASE_CNAME, type: 'double', width: 6 };
+
+    tPF.tableData.columns.push(phaseC);          // add phase column
+
+    tPF.tableData.data.forEach((d, index) => {   // add phase value (in string) to each data
+
+        tPF.tableData.data[index].push(getPhase(parseFloat(d[tIdx]), parseFloat(tzero), parseFloat(period)));
+    });
+
+    const fluxCols = get(getLayouInfo(), [LC.MISSION_DATA, LC.META_FLUX_NAMES]);
+
+    locateTableColumns(tPF, [timeName, LC.PHASE_CNAME,...fluxCols], 0);
+    return tPF;
+}
+
+
+/**
+ * @summary change the content of each row of the table by relocating some columns
+ *          to be after or replace certain column
+ * @param {TableModel} tbl
+ * @param {array|string} srcCols
+ * @param {number|string} targetCol
```

Comment:
just thought this parameter could be a column name or a column index meaning a table column that srcCols are relocated to.